#PMN_GCv2
---
#author: "Adonis D'Mello"
#output: PDFs/Txt files
---
#Packages

```{r}
require(cowplot)
require(dendextend)
require(edgeR)
require(FactoMineR)
require(ggdendro)
require(ggplot2)
require(gplots)
require(gridExtra)
require(gtools)
require(pvclust)
require(reshape)
require(vegan)
require(WGCNA)
require(ggfortify)
require(ggrepel)
require(GGally)
require(DESeq2)
require(pathview)
require(reshape2)
require(KEGGprofile)
require(UpSetR)
require(biomaRt)
require(tidyr)
require(factoextra)
library(png)
library(grid)
library(gridExtra)
library(ggpubr)
require(stringr)
```

#Functions
```{r}
get_heatmap_separators <- function(vector){
  sep <- c()
  for(i in 2:length(unique(vector))){
    sep[length(sep) + 1] <- min(which(vector == unique(vector)[i])) - 1
  }
  return(sep)
}

merge.all <- function(x, ..., by = "row.names") {
  L <- list(...)
  for (i in seq_along(L)) {
    x <- merge(x, L[[i]], by = by, all=TRUE)
    rownames(x) <- x$Row.names
    x$Row.names <- NULL
  }
  return(x)
}

get_dendro_structure <- function(result){
  structure <- hang.dendrogram(as.dendrogram(result$hclust))
  structure <- capture.output(str(structure))
  structure <- structure[grepl("leaf", structure)]
  structure <- as.numeric(as.character(substr(structure, regexpr("h=", structure ) + 3, regexpr("  )", structure))))
  return(structure)
}

get_dendro_data <- function(result){
  dendro.data <- dendro_data(result$hclust)
  dendro.data <- dendro.data$segments[which(dendro.data$segments$y == dendro.data$segments$yend),]
  for(i in 1:nrow(dendro.data)){
    dendro.data$minx[i] <- min(c(dendro.data$x[i], dendro.data$xend[i]))
  }
  dendro.data <- dendro.data[order(as.numeric(as.character(dendro.data$y)), as.numeric(as.character(dendro.data$minx))),]
  return(dendro.data)
}

get_dendro_bootstraps <- function(dendro_data){
  bootstrap.positions <- as.data.frame(matrix(nrow = length(dendro_data$y[duplicated(dendro_data$y)]),
                                              ncol = 2))
  for(i in 1:length(dendro_data$y[duplicated(dendro_data$y)])){
    dendro_data.subset <- dendro_data[which(dendro_data$y == dendro_data$y[duplicated(dendro_data$y)][i]),]
    bootstrap.positions[i,1] <- unique(dendro_data.subset$x)
    bootstrap.positions[i,2] <- unique(dendro_data.subset$y)
  }
  return(bootstrap.positions)
}

query_upset <- function(set, queries) {
  if (length(queries) ==1) {
    intersect <- set[,queries,drop = F]
    intersect$tmp = 0
    intersect <- intersect[rowSums(intersect)==1,]
  } else {
    intersect <- set[,queries]
    intersect <- intersect[rowSums(intersect) == length(queries),]
  }
  tmp <- set[rownames(intersect),]
  tmp <- tmp[rowSums(tmp) == length(queries),]
  return(tmp)
}

```


#Directories (Change as needed)
```{r}
input_directory <- "/Users/admello/OneDrive - University of Maryland School of Medicine/Desktop/PGAMR_paper/input/"
output_directory <- "/Users/admello/OneDrive - University of Maryland School of Medicine/Desktop/PGAMR_paper/output_2024 - Copy/"
```

#Inputs (Remove comments from only 1 Block at a time per run)

```{r}

#Input Block 1
# counts.path <- paste0(input_directory,"/130_NCBI_counts.matrix")
# groups.path <- paste0(input_directory,"/130_groups.txt")
# dir.create(file.path(output_directory, "130"))
# output_dir <- paste0(output_directory,"/130")
# comparisons <- c("130+PMN_1h_vs_130_1h","130+PMN_1h_vs_130+PMN_0h")

#Input Block 2
# counts.path <- paste0(input_directory,"/H041_NCBI_counts.matrix")
# groups.path <- paste0(input_directory,"/H041_groups.txt")
# dir.create(file.path(output_directory, "H041"))
# output_dir <- paste0(output_directory,"/H041")
# comparisons <- c("H041+PMN_1h_vs_H041_1h","H041+PMN_1h_vs_H041+PMN_0h")

#Input Block 3
# counts.path <- paste0(input_directory,"/PMN_counts.matrix")
# groups.path <- paste0(input_directory,"/PMN_groups.txt")
# dir.create(file.path(output_directory, "PMN"))
# output_dir <- paste0(output_directory,"/PMN")
# comparisons <- c("PMN_1h_vs_PMN_0h","130+PMN_1h_vs_130+PMN_0h","H041+PMN_1h_vs_H041+PMN_0h","130+PMN_1h_vs_PMN_1h","H041+PMN_1h_vs_PMN_1h")


#Input Block 4
counts.path <- paste0(input_directory,"/Core_counts_overhaul_ncbi.txt")
groups.path <- paste0(input_directory,"/Core_groups_overhaul.txt")
dir.create(file.path(output_directory, "GCcore"))
output_dir <- paste0(output_directory,"/GCcore")
comparisons <-c("H041+PMN_1h_vs_H041_1h","H041+PMN_1h_vs_H041+PMN_0h","130+PMN_1h_vs_130_1h","130+PMN_1h_vs_130+PMN_0h")

```

#Read inputs

```{r}
counts <- read.delim(counts.path, header = T, row.names = 1)
design <- read.delim(groups.path, header = T)

#Aquiring colors 
grp.col <- design[,ncol(design)-1]
grp.pch <- design[,ncol(design)]
```

#Deseq2 normalization

```{r}
#Normalization
if (length(colnames(design)) > 4) {
  dds <- DESeqDataSetFromMatrix(countData = floor(counts), colData = design, design = ~ donor + condition )
} else {
  dds <- DESeqDataSetFromMatrix(countData = counts, colData = design, design = ~ condition )
}
dds <- estimateSizeFactors(dds)
dds <- estimateDispersions(dds)
counts.vsd <- as.data.frame(getVarianceStabilizedData(dds))

write.table(counts.vsd, file = paste0(output_dir,"/Vst_counts.txt"), append = F, row.names = T,sep = "\t")
```

#PCA_allsamples

```{r}

pca <- prcomp(t(counts.vsd))
df_out <- as.data.frame(pca$x)
percentage <- round(pca$sdev^2 / sum(pca$sde^2) * 100, 2)
percentage <- paste( colnames(df_out), "(", paste(as.character(percentage), "% )", sep="") )

  pca.col <- unique(cbind(as.data.frame(design$condition),grp.col))[,2]
  pca.col <- factor(grp.col , levels = as.vector(pca.col))
  pca.groups <- factor(paste0(design$strain,design$condition), levels = unique(as.vector(paste0(design$strain,design$condition))))
  pca.shape <- factor(design$donor, levels =unique(design$donor))
  pca.pch <- unique(cbind(as.data.frame(design$donor),grp.pch))
  rownames(pca.pch) = pca.pch$`design$donor`
  pca.pch <- pca.pch[levels(pca.shape),]$grp.pch
  shp.text <- "Donor"

pca.plot <- ggplot(df_out, aes(x = PC1, y = PC2)) + geom_point(aes(fill = pca.groups, 
            shape=pca.shape),size = 5,show.legend = TRUE) + labs(title = "", fill = "Samples", shape=shp.text , x=percentage[1] , y=percentage[2])	+ 
              guides(fill=guide_legend(override.aes=list(shape=21))) + 
            scale_shape_manual(values = pca.pch) + scale_fill_manual(values = levels(pca.col)) + theme_bw()  

print(pca.plot)

pdf(paste0(output_dir,"/pca_plot.pdf"),
    height=8,
    width=8)
print(pca.plot)
dev.off()

pdf(paste0(output_dir,"/pca_scree.pdf"),
    height=5,
    width=8)
fviz_eig(pca)
dev.off()

```

#PCA T1s only

```{r}
design2 <- design[ design$ID %in% grep("_T0_",colnames(counts),value = T,invert = T),]
counts.vsd2 <- counts.vsd[,grep("*_T0_*",colnames(counts.vsd),value = T,invert = T)]
grp.col2 <- design2[,ncol(design2)-1]
grp.pch2 <- design2[,ncol(design2)]

pca <- prcomp(t(counts.vsd2))
df_out <- as.data.frame(pca$x)
percentage <- round(pca$sdev^2 / sum(pca$sde^2) * 100, 2)
percentage <- paste( colnames(df_out), "(", paste(as.character(percentage), "% )", sep="") )

  pca.col <- unique(cbind(as.data.frame(design2$condition),grp.col2))[,2]
  pca.col <- factor(grp.col2 , levels = as.vector(pca.col))
  pca.groups <- factor(paste0(design2$strain,design2$condition), levels = unique(as.vector(paste0(design2$strain,design2$condition))))
  pca.shape <- factor(design2$donor, levels =unique(design2$donor))
  pca.pch <- unique(cbind(as.data.frame(design2$donor),grp.pch2))
  rownames(pca.pch) = pca.pch$`design2$donor`
  pca.pch <- pca.pch[levels(pca.shape),]$grp.pch2
  shp.text <- "Donor"

pca.plot <- ggplot(df_out, aes(x = PC1, y = PC2)) + geom_point(aes(fill = pca.groups, 
            shape=pca.shape),size = 5,show.legend = TRUE) + labs(title = "", fill = "Samples", shape=shp.text , x=percentage[1] , y=percentage[2])	+ 
              guides(fill=guide_legend(override.aes=list(shape=21))) + 
            scale_shape_manual(values = pca.pch) + scale_fill_manual(values = levels(pca.col)) + theme_bw()  

print(pca.plot)

pdf(paste0(output_dir,"/pca_plot_T1s.pdf"),
    height=8,
    width=8)
print(pca.plot)
dev.off()

pdf(paste0(output_dir,"/pca_scree_T1s.pdf"),
    height=5,
    width=8)
fviz_eig(pca)
dev.off()

```


#Dendogram 

```{r}
dendrogram <- as.data.frame(counts.vsd)
result <- pvclust(dendrogram, method.dist="cor", method.hclust="average", nboot=100)

structure <- get_dendro_structure(result)
dendro.data <- get_dendro_data(result)
bootstrap.positions <- get_dendro_bootstraps(dendro.data)
points.df <- as.data.frame(cbind(seq(1,length(structure),1),
                                 structure))
  dendrogroups <- factor(design$condition[result$hclust$order], levels = unique(sort(as.vector(design$condition))))
  dendrocol <- unique(design[,c("condition","color")])
  dendrocol <- dendrocol[match(levels(dendrogroups), dendrocol$condition),][,2] 
  dendroshape <- factor(design$donor[result$hclust$order], levels = unique(sort(as.vector(design$donor))))
  dendropch <- unique(design[,c("donor","pch")])
  dendropch <- dendropch[match(levels(dendroshape), dendropch$donor),][,2] 
  shp.text <- "Donors"

dendrogram.plot <- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+
  geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups ,shape = dendroshape, fill=dendrogroups), size = 3)+ 
  labs(title = "B", x = "", y = "", color = "Samples", fill = "Samples", 
       shape = shp.text)+ scale_color_manual(values = dendrocol)+ scale_fill_manual(values =dendrocol)+
  scale_shape_manual(values = dendropch)+
  guides(colour = guide_legend(ncol = 1))+
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1, size = 7),
        axis.text.y = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())

for(i in 1:length(result$edges$bp)){
  text <- round(result$edges$bp[i] * 100,0)
  dendrogram.plot <- dendrogram.plot + annotate("text", label = text, x=bootstrap.positions[i,1] + 0.2, y=bootstrap.positions[i,2] + 0.002, size = 2)
}

print(dendrogram.plot)

pdf(paste0(output_dir,"/dendrogram_plot.pdf"),
    height=9,
    width=15)
print(dendrogram.plot)
dev.off()

```

#Deseq Independant Filtered Rarefy

```{r}

dds <- DESeq(dds, betaPrior = F)
contrast.list <- data.frame(Contrast=comparisons)
torarefy <- c()

for (i in contrast.list$Contrast) {
  pair <- as.data.frame(strsplit(as.character(i),'_vs_'))
	DE.genes <- as.data.frame(results(dds, contrast=c("condition",as.character(pair[1,]),as.character(pair[2,])), independentFiltering = TRUE, alpha=0.05))
	DE.genes <- na.omit(DE.genes)
	rareall <- as.data.frame(c(rep_len(1,length.out = length(row.names(DE.genes)))),row.names = rownames(DE.genes) )
	torarefy <- merge.all(torarefy,rareall)
}

rarefy.counts <- round(counts[rownames(torarefy),],0)

raremax <- round(min(rowSums(t(rarefy.counts))),0)
srare <- rarefy(t(rarefy.counts),raremax)

rarefy.raw.df <- rarecurve(t(rarefy.counts), step = round(raremax/1000,0), sample = raremax)

rarefy.df <- as.data.frame(matrix(nrow = 0,
                                  ncol = 5))
rarefy.points.df <- rarefy.df
for(i in 1:length(rarefy.raw.df)){
  steps <- as.numeric(gsub("N","",names(rarefy.raw.df[[i]])))
  detected_genes <- as.numeric(rarefy.raw.df[[i]])
  rarefy.df <- as.data.frame(rbind(rarefy.df,
                                   cbind(as.numeric(steps),as.numeric(detected_genes),as.character(design[i,1]),as.character(design[i,3]),design[i,4])))
  rarefy.points.df <- as.data.frame(rbind(rarefy.points.df,
                                          cbind(as.numeric(max(steps)),as.numeric(max(detected_genes)),as.character(design[i,1]),as.character(design[i,3],design[i,4]))))
  
}
rarefy.plot <- ggplot()+
  geom_line(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+
  geom_point(mapping=aes(x=as.numeric(as.character(rarefy.points.df[,1])), y=as.numeric(as.character(rarefy.points.df[,2])),group=rarefy.points.df[,3],color=rarefy.points.df[,4]),size = 3)+
  #guides(color = F,shape = F)+
  scale_color_manual(values = levels(factor(grp.col, levels = unique(grp.col))) )+
  labs(x="Reads mapping to protein-coding genes", y="Genes detected", color = "Sample") + xlim(0,10000000)+
  theme_bw() 

pdf(paste0(output_dir,"/rarefication_plot.pdf"),
    height=5,
    width=8)
grid.arrange(rarefy.plot)
dev.off()


```

#DEseq2 Differentially expressed genes

```{r}
dds <- DESeq(dds, betaPrior = F)

contrast.list <- data.frame(Contrast=comparisons)

for (i in contrast.list$Contrast) {

  	pair <- as.data.frame(strsplit(as.character(i),'_vs_'))
	DE.genes <- as.data.frame(results(dds, contrast=c("condition",as.character(pair[1,]),as.character(pair[2,])), independentFiltering = TRUE, alpha=0.05))
	DE.genes <- na.omit(DE.genes)
	rownames(DE.genes) <- gsub(".p01","",rownames(DE.genes))
  heading<- paste(as.character(pair[1,])," vs ",as.character(pair[2,])," (Abs(LFC) > 1"," and FDR < 0.05",")", sep="")
	DE.genes$Significant = as.factor(DE.genes$padj <= 0.05 & abs(DE.genes$log2FoldChange) >= 1)
  Significant <- c(DE.genes$Significant)
  write.table(data.frame("ID"=rownames(DE.genes),DE.genes), file = paste0(output_dir,"/",i,"_unfiltered_degenes.txt"), append = F, row.names = F,sep = "\t", quote = FALSE)
	DE.genes <- DE.genes[DE.genes$padj <= 0.05,]
	write.table(data.frame("ID"=rownames(DE.genes),DE.genes), file = paste0(output_dir,"/",i,"_FDR_degenes.txt"), append = F, row.names = F,sep = "\t", quote = FALSE)
	genes_fdr <- as.data.frame(c(rep_len(1,length.out = length(row.names(DE.genes)))),row.names = rownames(DE.genes) )
	DE.genes <- DE.genes[abs(DE.genes$log2FoldChange) >=1,]
	write.table(data.frame("ID"=rownames(DE.genes),DE.genes), file = paste0(output_dir,"/",i,"_FDRLFC_degenes.txt"), append = F, row.names = F,sep = "\t", quote = FALSE)
}
```

#Heatmaps 130 (for Input Block 1 Only)
#Swap between main figure heatmap database and supplemental db by commenting in/out lines 385 & 386 (first 2 lines from the below block)

```{r}

regdb <- readLines(paste0(input_directory,"/GCVenn_130_2024.db"))
#regdb <- readLines(paste0(input_directory,"/GCVenn_130_supp_2024.db"))

isHdr <- grepl("#", regdb)
grp <- regdb[isHdr][cumsum(isHdr)]
Read <- function(x) read.table(text = x, sep = "\t", fill = TRUE, comment = "#")
regs <- Map(Read, split(regdb, grp))

counts.vsd <-read.delim(paste0(output_directory,"/130/VST_counts.txt"), header = T, row.names = 1)
ord <- c(1,2,7,8,13,14,3,4,9,10,15,16,5,6,11,12,17,18)
#ord <- c(1,2,5,6,9,10,3,4,7,8,11,12)
counts.vsd <- counts.vsd[,ord]
counts.vsd <- counts.vsd[,grep("*_T0_*",colnames(counts.vsd),value = T,invert = T)]


for (i in c(1:length(regs))){
#colsep <- get_heatmap_separators(as.character(c("X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h")))
colsep <- get_heatmap_separators(as.character(c("X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h")))
hmcol <- colorRampPalette(c("gold3","white","darkorchid3"))(75)
pdf(paste0(output_dir,"/Heatmap_",names(regs[i]),".pdf"),width=8, height=25)#,paper = "a4")

#aka2 = data.frame(Condition = factor(c("X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_0h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h")))
aka2 = data.frame(Condition = factor(c("X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h","X130PMN_1h")))

rownames(aka2)<-colnames(counts.vsd)
#aka3 = list(Condition = c(X130_1h = "skyblue3",X130PMN_0h = "lightskyblue1",X130PMN_1h = "darkblue"))
aka3 = list(Condition = c(X130_1h = "skyblue3",X130PMN_1h = "darkblue"))

    rownames(regs[[i]]) <- regs[[i]]$V2
    regs[[i]] <- regs[[i]][mixedsort(regs[[i]]$V2),]

    rowns <-paste0(regs[[i]]$V2,":",regs[[i]]$V1,"\t",regs[[i]]$V3)
      rowns <- rowns[!grepl('None', rowns)]
      if (length(rownames(na.omit(t(scale(t(counts.vsd[regs[[i]]$V1,])))))) == length(rowns)) {
pheatmap::pheatmap(na.omit(t(scale(t(counts.vsd[regs[[i]]$V1,])))),color = hmcol,cellwidth = 10,cellheight = 10,cluster_rows = F,labels_row = rowns ,
  cluster_cols = F,annotation_col = aka2,annotation_colors = aka3[1], main = names(regs[i]))
      }
dev.off()
}
```

#Heatmaps H041 (for Input Block 2 Only)
#Swap between main figure heatmap database and supplemental db by commenting in/out lines 431 & 432 (first 2 lines from the below block)

```{r}

regdb <- readLines(paste0(input_directory,"/GCVenn_H041_2024.db"))
#regdb <- readLines(paste0(input_directory,"/GCVenn_H041_supp_2024.db"))

isHdr <- grepl("#", regdb)
grp <- regdb[isHdr][cumsum(isHdr)]
Read <- function(x) read.table(text = x, sep = "\t", fill = TRUE, comment = "#")
regs <- Map(Read, split(regdb, grp))

counts.vsd <-read.delim(paste0(output_directory,"/H041/VST_counts.txt"), header = T, row.names = 1)
ord <- c(1,2,7,12,3,4,8,9,13,14,5,6,10,11,15,16)
#ord <- c(1,2,5,8,3,4,6,7,9,10)
counts.vsd <- counts.vsd[,ord]
counts.vsd <- counts.vsd[,grep("*_T0_*",colnames(counts.vsd),value = T,invert = T)]

for (i in c(1:length(regs))){
#colsep <- get_heatmap_separators(as.character(c("H041_1h","H041_1h","H041_1h","H041_1h","H041PMN_0h","H041PMN_0h","H041PMN_0h","H041PMN_0h","H041PMN_0h","H041PMN_0h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h")))
colsep <- get_heatmap_separators(as.character(c("H041_1h","H041_1h","H041_1h","H041_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h")))

hmcol <- colorRampPalette(c("gold3","white","darkorchid3"))(75)
pdf(paste0(output_dir,"/Heatmap_",names(regs[i]),".pdf"),width=8, height=25)#,paper = "a4")

aka2 = data.frame(Condition = factor(c("H041_1h","H041_1h","H041_1h","H041_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h","H041PMN_1h")))
rownames(aka2)<-colnames(counts.vsd)
#aka3 = list(Condition = c(H041_1h = "orangered3",H041PMN_0h = "orangered1",H041PMN_1h = "orangered4"))
aka3 = list(Condition = c(H041_1h = "orangered3",H041PMN_1h = "orangered4"))

    rownames(regs[[i]]) <- regs[[i]]$V2
    regs[[i]] <- regs[[i]][mixedsort(regs[[i]]$V2),]

    rowns <-paste0(regs[[i]]$V2,":",regs[[i]]$V1,"\t",regs[[i]]$V3)
      rowns <- rowns[!grepl('None', rowns)]
      if (length(rownames(na.omit(t(scale(t(counts.vsd[regs[[i]]$V1,])))))) == length(rowns)) {
pheatmap::pheatmap(na.omit(t(scale(t(counts.vsd[regs[[i]]$V1,])))),color = hmcol,cellwidth = 10,cellheight = 10,cluster_rows = F,labels_row = rowns ,
  cluster_cols = F,annotation_col = aka2,annotation_colors = aka3[1], main = names(regs[i]))
      }
dev.off()
}

```

#Heatmaps PMN (for Input Block 3 Only)

```{r}

regdb <- readLines(paste0(input_directory,"/GCVenn_Host_2024.db"))
isHdr <- grepl("#", regdb)
grp <- regdb[isHdr][cumsum(isHdr)]
Read <- function(x) read.table(text = x, sep = "\t", fill = TRUE, comment = "#")
regs <- Map(Read, split(regdb, grp))

counts.vsd <-read.delim(paste0(output_directory,"/PMN/VST_counts.txt"), header = T, row.names = 1)
# ord <- c("D55_PMN_130_T0_20180201","D55_PMN_130_T0_20181106","D60_PMN_130_T0_20180208","D60_PMN_130_T0_20181113","D6_PMN_130_T0_20180131","D6_PMN_130_T0_20181108","D55_PMN_130_T1_20180201","D55_PMN_130_T1_20181106","D60_PMN_130_T1_20180208","D60_PMN_130_T1_20181113","D6_PMN_130_T1_20180131","D6_PMN_130_T1_20181108","D55_PMN_H041_T0_20180201","D55_PMN_H041_T0_20190110","D60_PMN_H041_T0_20180208","D60_PMN_H041_T0_20181212","D6_PMN_H041_T0_20180131","D6_PMN_H041_T0_20190122","D55_PMN_H041_T1_20180201","D55_PMN_H041_T1_20190110","D60_PMN_H041_T1_20180208","D60_PMN_H041_T1_20181212","D6_PMN_H041_T1_20180131","D6_PMN_H041_T1_20190122","D55_PMN_alone_T0_20180920","D55_PMN_alone_T0_20190416","D60_PMN_alone_T0_20190416","D6_PMN_alone_T0_20180802","D6_PMN_alone_T0_20190416","D55_PMN_alone_T1_20180920","D55_PMN_alone_T1_20190416","D60_PMN_alone_T1_20190416","D6_PMN_alone_T1_20180802","D6_PMN_alone_T1_20190416")
# grps <- c("X130_PMN_0h","X130_PMN_0h","X130_PMN_0h","X130_PMN_0h","X130_PMN_0h","X130_PMN_0h","X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","H041_PMN_0h","H041_PMN_0h","H041_PMN_0h","H041_PMN_0h","H041_PMN_0h","H041_PMN_0h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","PMN_0h","PMN_0h","PMN_0h","PMN_0h","PMN_0h","PMN_1h","PMN_1h","PMN_1h","PMN_1h","PMN_1h")

ord <- c("D55_PMN_130_T1_20180201","D55_PMN_130_T1_20181106","D60_PMN_130_T1_20180208","D60_PMN_130_T1_20181113","D6_PMN_130_T1_20180131","D6_PMN_130_T1_20181108","D55_PMN_H041_T1_20180201","D55_PMN_H041_T1_20190110","D60_PMN_H041_T1_20180208","D60_PMN_H041_T1_20181212","D6_PMN_H041_T1_20180131","D6_PMN_H041_T1_20190122","D55_PMN_alone_T1_20180920","D55_PMN_alone_T1_20190416","D60_PMN_alone_T1_20190416","D6_PMN_alone_T1_20180802","D6_PMN_alone_T1_20190416")

grps <- c("X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","X130_PMN_1h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","H041_PMN_1h","PMN_1h","PMN_1h","PMN_1h","PMN_1h","PMN_1h")

counts.vsd <- counts.vsd[,ord]

for (i in c(1:length(regs))){

colsep <- get_heatmap_separators(as.character(grps))
hmcol <- colorRampPalette(c("gold3","white","darkorchid3"))(75)
pdf(paste0(output_dir,"/Heatmap2_",names(regs[i]),".pdf"),width=10, height=25)#,paper = "a4")

aka2 = data.frame(Condition = factor(grps))
rownames(aka2)<-colnames(counts.vsd)

# aka3 = list(Condition = c(X130_PMN_0h = "lightskyblue1",X130_PMN_1h = "darkblue",H041_PMN_0h = "orangered1",H041_PMN_1h = "orangered4",PMN_0h = "gray", PMN_1h ="black"))
aka3 = list(Condition = c(X130_PMN_1h = "darkblue",H041_PMN_1h = "orangered4", PMN_1h ="black"))

aka3[1]

    rownames(regs[[i]]) <- regs[[i]]$V2
    rowns <-paste0(regs[[i]]$V1,"\t",regs[[i]]$V3)
      rowns <- rowns[!grepl('None', rowns)]
    #   
    # x50s <- counts.vsd[regs[[i]]$V1,]
    # x50s$mean <- rowMeans(x50s)
    # #sorted by vst avg in each subgroup
    # x50s2<-rbind(x50s[order(-x50s[c(1:20),]$mean),],x50s[order(-x50s[c(21:34),]$mean)+20,])
    # x50s2<-rbind(x50s2,x50s[order(-x50s[c(35:43),]$mean)+34,])
    # x50s2<-rbind(x50s2,x50s[order(-x50s[c(44:50),]$mean)+43,])
    # x50s2$mean <- NULL
    # 
    # rowns <- rowns[match(rownames(counts.vsd[regs[[i]]$V1,]),rownames(x50s2))]
    # setdiff(rownames(counts.vsd[regs[[i]]$V1,]),rownames(x50s2))

      if (length(rownames(na.omit(t(scale(t(counts.vsd[regs[[i]]$V1,])))))) == length(rowns)) {
pheatmap::pheatmap(na.omit(t(scale(t(counts.vsd[regs[[i]]$V1,])))),color = hmcol,cellwidth = 10,cellheight = 10,cluster_rows = F,labels_row = rowns ,
  cluster_cols = F,annotation_col = aka2,annotation_colors = aka3[1], main = names(regs[i]))
      }    
dev.off()
}

```











